src/main/java/org/jabref/logic/ai/embeddings/MVStoreEmbeddingStore.java

```diff
@@ -0,0 +1,163 @@
+        add(id, embedding);
+
+        contentsMap.put(id, textSegment.text());
+
+        String linkedFile = textSegment.metadata().getString("linkedFile");
+        if (linkedFile != null) {
+            fileMap.put(id, linkedFile);
+        }
+
+        return id;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        return embeddings.stream().map(this::add).toList();
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        return IntStream.range(0, embeddings.size()).mapToObj(i -> add(embeddings.get(i), embedded.get(i))).toList();
+    }
+
+    @Override
+    public void remove(String id) {
+        embeddingsMap.remove(id);
+        contentsMap.remove(id);
+        fileMap.remove(id);
+    }
+
+    @Override
+    public void removeAll(Filter filter) {
+        List<String> idsToRemove = applyFilter(filter).toList();
+        idsToRemove.forEach(this::remove);
+    }
+
+    @Override
+    public void removeAll() {
+        embeddingsMap.clear();
+        contentsMap.clear();
+        fileMap.clear();
+    }
+
+    @Override
+    public EmbeddingSearchResult<TextSegment> search(EmbeddingSearchRequest request) {
+        // Source: InMemoryEmbeddingStore.
+
+        Comparator<EmbeddingMatch<TextSegment>> comparator = comparingDouble(EmbeddingMatch::score);
+        PriorityQueue<EmbeddingMatch<TextSegment>> matches = new PriorityQueue<>(comparator);
+
+        Filter filter = request.filter();
+
+        applyFilter(request.filter()).forEach(id -> {
+            Embedding embedding = new Embedding(embeddingsMap.get(id));
+
+            double cosineSimilarity = CosineSimilarity.between(embedding, request.queryEmbedding());
+
+            double score = RelevanceScore.fromCosineSimilarity(cosineSimilarity);
+
+            if (score >= request.minScore()) {
+                matches.add(new EmbeddingMatch<>(score, id, embedding, new TextSegment(contentsMap.get(id), new Metadata())));
+
+                if (matches.size() > request.maxResults()) {
+                    matches.poll();
+                }
+            }
+        });
+
+        List<EmbeddingMatch<TextSegment>> result = new ArrayList<>(matches);
+        result.sort(comparator);
+        Collections.reverse(result);
+
+        return new EmbeddingSearchResult<>(result);
+    }
+
+    @Override
+    public void removeAll(Collection ids) {
+        embeddingsMap.entrySet().removeIf(entry -> ids.contains(entry.getKey()));
+        contentsMap.entrySet().removeIf(entry -> ids.contains(entry.getKey()));
+        fileMap.entrySet().removeIf(entry -> ids.contains(entry.getKey()));
+    }
+
+    private Stream<String> applyFilter(@Nullable Filter filter) {
+        return switch (filter) {
+            case null -> embeddingsMap.keySet().stream();
+
+            case IsIn isInFilter when Objects.equals(isInFilter.key(), "linkedFile") ->
```

Comment:
When I saw the error messages correctly, this is the part where checkstyle fails.

Can you verify this locally with running checkstyle?


https://docs.oracle.com/en/java/javase/21/language/pattern-matching-switch-expressions-and-statements.html#GUID-A5C220F6-F70A-4FE2-ADB8-3B8883A67E8A 

I found related issues

https://github.com/checkstyle/checkstyle/issues/15044 
https://github.com/checkstyle/checkstyle/issues/15082